model/src/main/java/uk/gov/hmcts/payment/api/jpaaudit/listner/PaymentFeeLinkEntityListener.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.payment.api.jpaaudit.listner;
+
+import uk.gov.hmcts.payment.api.jpaaudit.model.AuditEventsType;
+import uk.gov.hmcts.payment.api.jpaaudit.model.PaymentAuditHistory;
+import uk.gov.hmcts.payment.api.model.CaseDetails;
+import uk.gov.hmcts.payment.api.model.PaymentFeeLink;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PostPersist;
+import javax.persistence.PostUpdate;
+import javax.transaction.Transactional;
+
+import static javax.transaction.Transactional.TxType.MANDATORY;
+import static uk.gov.hmcts.payment.api.jpaaudit.listner.Action.INSERTED;
+import static uk.gov.hmcts.payment.api.jpaaudit.listner.Action.UPDATED;
+import static uk.gov.hmcts.payment.api.jpaaudit.model.AuditEventsType.ORDER_CREATED;
+
+public class PaymentFeeLinkEntityListener {
+
+    @PostPersist()
+    public void postPersist(PaymentFeeLink target) {
+        perform(target, INSERTED, ORDER_CREATED);
+    }
+
+    @Transactional(MANDATORY)
+    private void perform(PaymentFeeLink target, Action action, AuditEventsType auditEventsType) {
+        EntityManager entityManager = BeanUtil.getBean(EntityManager.class);
+
+//        String ccdCaseNumber =target.getCaseDetails().iterator().next().getCcdCaseNumber();
+//        Will be used once all services onboarded orders endpoint
+        entityManager.persist(PaymentAuditHistory.paymentAuditHistoryWith()
+            .ccdCaseNo(target.getCcdCaseNumber())
```

Comment:
if target.getCcdCaseNumber() would work here ? I think Order should retrieve the ccd_case from CASE_DETAILS through ORDER_CASE

Reply:
I commented the line 29,30 which will be used once other controllers store data in the caseDetails table.
//        String ccdCaseNumber =target.getCaseDetails().iterator().next().getCcdCaseNumber();
//        Will be used once all services onboarded orders endpoint